### DISCUSSION TITLE
Support for unsupervised change detection

### DISCUSSION THREAD

**Discussion by blaz-r** [torchgeo/torchgeo#4013] [Ideas]
I think it would be beneficial to add unsupervised or even training-free methods for change detection. Lots of research is not working on this, including the open-vocabulary category.

In the past few months, we have received quite a few requests from some of our partners who want to do some quick CD but do not always have labels. TorchGeo currently supports only supervised models, with the exception of some pre-trained models that do not really transfer well to new domains. This makes me believe that such an addition would be quite useful for practical usage.

I think one pretty cool method is [AnyChange](https://arxiv.org/pdf/2402.01188), which is in https://github.com/Z-Zheng/pytorch-change-models, and it requires no additional training.
I'm also asking, as we have our own UCD method, [MaSoN](https://blaz-r.github.io/mason_ucd/), which is unsupervised but requires about 1000 training iterations. This one, specifically, could fit inside the current trainer with some modifications to how masks are handled (MaSoN generates masks synthetically).

**Comment by adamjstewart** [comment]
AnyChange looks like a vision–language foundation model pre-trained with self-supervised learning, not an unsupervised model. Less familiar with UCD. Not sure how well these will generalize to new regions with zero-shot learning. Happy to add them though.

In terms of adding these models to `torchgeo.tasks.ChangeDetection`, it should be just a few lines of code. If you don't have any labels in your dataset/datamodule, you would just skip straight to `predict` and skip the fit/test stages. Feel free to open PRs for these models! Maybe 1 PR for each model and a separate PR for each task you add it to.

**Reply by blaz-r** [reply]
Great. I will do that then, once I get some time 😄

**Reply by blaz-r** [reply]
@adamjstewart anhcange is part of torchange, which has an Apache licence. I'm not sure what the best way is to handle that, then. Can we still somehow include it? It is offered as a package, which I also thought would be maybe the simplest way to implement, but I'm not too happy, as I'm unsure how that'll work dependency-wise.

**Reply by adamjstewart** [reply]
We could add an optional dep on torchange and add all of those models to the CD task

**Reply by blaz-r** [reply]
Cool, I'll do that then.
